numba/np/arraymath.py

```diff
@@ -939,18 +939,18 @@ def nanvar_impl(a):
                 ssd += np.real(val * np.conj(val))
                 count += 1
         # np.divide() doesn't raise ZeroDivisionError
-        return np.divide(ssd, count)
+        return np.divide(ssd, (count - ddof))
 
     return nanvar_impl
 
 
 @overload(np.nanstd)
-def np_nanstd(a):
+def np_nanstd(a, ddof=0):
     if not isinstance(a, types.Array):
         return
 
-    def nanstd_impl(a):
-        return np.nanvar(a) ** 0.5
+    def nanstd_impl(a, ddof=ddof):
```

Comment:
I think the test failures come down to this:
```suggestion
    def nanstd_impl(a, ddof=0):
```
the outer scope function arguments are in the "numba type" domain, the inner are in the "numba compilation" domain, by using the outer scope definition here it's asking the compiler to compile a type opposed to an instance of the type. There's a user guide here which explains some of this https://numba.readthedocs.io/en/stable/extending/overloading-guide.html hope this helps?